### DISCUSSION TITLE
Where is the Function Event Data?

### DISCUSSION THREAD

**Discussion by jbdujardin** [appwrite/appwrite#4437] [Q&A] [answered]
### 👟 Reproduction steps

- Create new Function (example : `on-team-created` with Runtime `dart-2.17`)
- Add the event for this function (example : `teams.*.create`)
- Just send the payload on the function body : 
```
Future<void> start(final req, final res) async {
  res.send(req.payload);
}
```
- Trigger the event (example : Create new team with the console)

### 👍 Expected behavior

We have data on the payload.

### 👎 Actual Behavior

The payload is empty.

### 🎲 Appwrite version

Version 1.0.x

### 💻 Operating system

MacOS

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by akarsh-jain-790** [comment]
Hey, can u assign this issue to me?

**Comment by stnguyen90** [comment, marked as answer]
@jbdujardin, 

`req.payload` contains the data sent when a function is executed via the [Create Execution API](https://appwrite.io/docs/client/functions#functionsCreateExecution). 

Event data can be found in `req.variables['APPWRITE_FUNCTION_EVENT_DATA']`. Please refer to the [Function Variables docs](https://appwrite.io/docs/functions#functionVariables) for more info.

If this answers your question, please mark this response as the answer 🙏

**Reply by jbdujardin** [reply]
Hey! Thank you and sorry for my bad question, it's ok with `req.variables['APPWRITE_FUNCTION_EVENT_DATA']`
